snuba/datasets/transactions_processor.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+
+import ipaddress
+import uuid
+
+from snuba.processor import (
+    _as_dict_safe,
+    MessageProcessor,
+    _ensure_valid_date,
+    _unicodify
+)
+from snuba.datasets.events_processor import (
+    enforce_retention,
+    extract_base,
+    extract_extra_contexts,
+    extract_extra_tags,
+    extract_user,
+)
+
+
+class TransactionsMessageProcessor(MessageProcessor):
+    PROMOTED_TAGS = [
+        "environment",
+        "sentry:release",
+        "sentry:user",
+        "sentry:dist",
+    ]
+
+    def __extract_timestamp(self, field):
+        timestamp = _ensure_valid_date(datetime.fromtimestamp(field))
+        milliseconds = int(timestamp.microsecond / 1000)
+        return (timestamp, milliseconds)
+
+    def process_message(self, message, metadata=None):
+        action_type = self.INSERT
+        processed = {'deleted': 0}
+        if not isinstance(message, (list, tuple)) and len(message) >= 2:
+            return None
+        version = message[0]
+        if version not in (0, 1, 2):
+            return None
+        type_, event = message[1:3]
+        if type_ != 'insert':
+            return None
+
+        data = event.get("data", {})
+        event_type = data.get("type")
+        if event_type != "transaction":
+            return None
+        base = extract_base(event)
+        enforce_retention(
+            event,
+            datetime.fromtimestamp(data['timestamp']),
+            base['retention_days'],
+            base['project_id'],
+        )
+        processed.update(base)
+
+        transaction_ctx = data["contexts"]["trace"]
+        trace_id = transaction_ctx["trace_id"]
+        processed["event_id"] = str(uuid.UUID(processed["event_id"]))
+        processed["trace_id"] = str(uuid.UUID(trace_id))
+        processed["span_id"] = int(transaction_ctx["span_id"], 16)
+        processed["transaction_op"] = _unicodify(transaction_ctx.get("op", ""))
+        processed["transaction_name"] = _unicodify(data["transaction"])
+
+        processed["start_ts"], processed["start_ms"] = self.__extract_timestamp(
+            data["start_timestamp"],
+        )
+        processed["finish_ts"], processed["finish_ms"] = self.__extract_timestamp(
+            data["timestamp"],
+        )
+
+        processed['platform'] = _unicodify(event['platform'])
+
+        tags = _as_dict_safe(data.get('tags', None))
+        extract_extra_tags(processed, tags)
+
+        promoted_tag = {col: _unicodify(tags[col])
```

Comment:
```suggestion
        promoted_tags = {col: _unicodify(tags[col])
```

The references below will also need to change.